### ISSUE TITLE
Debug Protocol: support 'restart frame'

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode-debugadapter-node#43]
Add protocol support for a `restartFrame` request.
A single argument `frameId` specifies the frame to restart.

**Comment by DavidKarlas** [comment]
This is similar to https://github.com/Microsoft/vscode-debugadapter-node/issues/28?
Maybe it could be merged into one command with addition of optional frame index parameter?
It might be confusing having to very similar commands unless I misunderstand this command...

**Comment by weinand** [comment]
@DavidKarlas good idea, I will consider your suggestion.
For now I have already added the `restartFrame` to the protocol, but the 1.11 version of the protocol is only a preview until the end of the sprint.

**Comment by weinand** [comment]
@DavidKarlas I've tried to merge 'restartFrame' into the 'setNextStatement' proposed in #28 but in the end I gave up because supporting different kinds of parameters (frameId vs. source location) on 'setNextStatement' was just adding complexity without gaining much: the UI for 'setNextStatement' is quite different from 'restartFrame'. The former works on individual statements in the source, the latter is more useful on call stack frames. So typically there is no shared code that has to deal with both cases.
